packages/date-picker/src/DatePicker/DatePicker.spec.tsx

```diff
@@ -3042,12 +3018,21 @@ describe('packages/date-picker', () => {
             await waitFor(() => expect(monthInput).toHaveValue('02'));
           });
 
-          test('if the resulting value is not valid, clears the input', async () => {
+          test('if the resulting value is incomplete and not valid, clears the input', async () => {
             const { monthInput } = renderDatePicker({});
             userEvent.type(monthInput, '0');
             userEvent.tab();
             await waitFor(() => expect(monthInput).toHaveValue(''));
           });
+
+          test('if the resulting value is not valid, formats the value', async () => {
```

Comment:
I think this test name could be more descriptive (maybe call out that the 2nd digit is inputted into the next input)